### PR TITLE
Installer container

### PR DESCRIPTION
**Pull request by stevemul**
Adding a template with which to create the imagestreams and buildconfig required for each pipeline, the Dockerfile for the openshift-installer container and updates to the README to use the template with.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,35 @@ OpenShift deployment code based on Tekton.
...
+
+```
+BUILDCONFIG_NAME - the name to give the BuildConfig
+OUTPUT_IMAGESTREAM_NAME - the name to give the ImageStream that will contain your output image
+BASE_IMAGESTREAM_NAME - the base image to use with your Dockerfile
+GIT_SOURCE - the GIT repo you'll pull the Dockerfile from
+GIT_BRANCH - the branch of the repo to use
+CONTEXT_DIR - the directory containing your Dockerfile in the Git repo
+SOURCE_IMAGE_FOR_BASE - the url to access the source image you wish to use as your base (this defaults to the Red Hat RHEL8 UBI minimal image)
+```
```
Comment (andrew-garner) [inline README.md:26]: Would be good to add the other three possible paremeters here, NAMESPACE, OUTPUT_IMAGE_STREAM_TAG and BASE_IMAGESTREAM_TAG, even if they have defaults.

File: README.md [hunk 2]
```diff
@@ -4,6 +4,35 @@ OpenShift deployment code based on Tekton.
...
+
+Additional variable are available to control the tags used on the input and output ImageStreams. Look in the template file for details, these currently default to latest.
+
+To use the template the following is an example that will work with the OpenShift CLI:
+
+```
+oc process -f  buildconfig_imagestream_templates.yaml -p BUILDCONFIG_NAME=test-buildconfig \
+	-p OUTPUT_IMAGESTREAM_NAME=output_imagestream_name \
```
Comment (andrew-garner) [inline README.md:34]: I'd find it useful to have examples of (mostly working) options. ie does the output ImageStream name need the docker:// prefix? Some of the cluster tasks need this and others don't but it's not documented anywhere.

**Review by andrew-garner — COMMENTED** [review]
Comments about comments!

File: README.md (hunk 2 above)
Comment (stevemul) [reply README.md:34]: That is a working example, it literally creates an output ImageStream called output_imagestream_name with this config applied.

**Review by k8soneill — DISMISSED** [review]
LGTM - assuming it works!

File: README.md (hunk 1 above)
Comment (stevemul) [reply README.md:26]: Updated and removed the unused namespace variable

**Review by andrew-garner — APPROVED** [review]
LGTM

**Review by k8soneill — APPROVED** [review]
Looks even gooder to me

**Pull request merged**